.github/workflows/test.yml

```diff
@@ -21,12 +21,33 @@ jobs:
         server:
         - mir_kiosk
         - ubuntu_frame
-        - confined_shell
         - mir_test_tools
-        - mir_demo_server
-        - gnome_shell
+        channel: [stable]
         include:
+        - server: ubuntu_frame
+          channel: 24/edge
+
+        - server: mir_test_tools
+          channel: 24/edge
+
+        - server: mir_demo_server
+          channel: null
+
+        - server: mir_demo_server
+          channel: null
+          ppa: mir-team/dev
+
+        - server: confined_shell
+          channel: beta
+
+        - server: confined_shell
+          channel: edge
+
+        - server: gnome_shell
+          channel: null
+
         - server: null
+          channel: null
           mark: self
```

Comment:
What does this do?

Reply:
This runs only the self tests, those marked with `@pytest.mark.self`. And measures coverage on them. So no server is involved - hence no channel.